core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -364,4 +364,62 @@ void shouldSucceedWhenCommitOnTestSourcesAfterReleaseDate() throws IOException,
             verify(probe).doApply(any(Plugin.class), any(ProbeContext.class));
         }
     }
+
+    @Test
+    void whenAFileIsCommittedBeforeAndAfterReleaseDateOnlyFilesAfterReleaseDateShouldShow() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final String scmLink = "https://test-server/jenkinsci/test-repo/test-folder";
+        final String pluginName = "test-plugin";
+
+        when(plugin.getReleaseTimestamp()).thenReturn(ZonedDateTime.now());
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+        when(plugin.getName()).thenReturn(pluginName);
+        when(plugin.getScm()).thenReturn(scmLink);
+
+        try (Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            Files.createFile(repository.resolve("pom.xml"));
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().minusDays(1).toInstant()));
+
+            git.add().addFilepattern("pom.xml").call();
+            git.commit().setMessage("Imports pom.xml file after commit date").setSign(false).setCommitter(committer).call();
+
+            final Path srcTestJava = Files.createDirectories(repository.resolve("src").resolve("main").resolve("java"));
+            Files.createFile(srcTestJava.resolve("TestClass.java"));
```

Comment:
Name is confusing because it's not a test class

```suggestion
            Files.createFile(srcTestJava.resolve("Hello.java"));
```